lumicks/pylake/force_calibration/calibration_models.py

```diff
@@ -99,3 +102,135 @@ def calibration_results(self, fc, diffusion_constant):
+
+
+class ActiveCalibrationModel(CalibrationModel):
+    """Model to fit data acquired during passive calibration.
+
+    The power spectrum calibration algorithm implemented here is based on [1]_, [2]_, [3]_, [4]_,
+    [5]_, [6]_.
+
+    References
+    ----------
+    .. [1] Berg-Sørensen, K. & Flyvbjerg, H. Power spectrum analysis for optical tweezers. Rev. Sci.
+           Instrum. 75, 594 (2004).
+    .. [2] Tolić-Nørrelykke, I. M., Berg-Sørensen, K. & Flyvbjerg, H. MatLab program for precision
+           calibration of optical tweezers. Comput. Phys. Commun. 159, 225–240 (2004).
+    .. [3] Hansen, P. M., Tolic-Nørrelykke, I. M., Flyvbjerg, H. & Berg-Sørensen, K.
+           tweezercalib 2.1: Faster version of MatLab package for precise calibration of optical
+           tweezers. Comput. Phys. Commun. 175, 572–573 (2006).
+    .. [4] Berg-Sørensen, K., Peterman, E. J. G., Weber, T., Schmidt, C. F. & Flyvbjerg, H. Power
+           spectrum analysis for optical tweezers. II: Laser wavelength dependence of parasitic
+           filtering, and how to achieve high bandwidth. Rev. Sci. Instrum. 77, 063106 (2006).
+    .. [5] Tolić-Nørrelykke, S. F, and Flyvbjerg, H, "Power spectrum analysis with least-squares
+           fitting: amplitude bias and its elimination, with application to optical tweezers and
+           atomic force microscope cantilevers." Review of Scientific Instruments 81.7 (2010)
+    .. [6] Tolić-Nørrelykke S. F, Schäffer E, Howard J, Pavone F. S, Jülicher F and Flyvbjerg, H.
+           Calibration of optical tweezers with positional detection in the back focal plane,
+           Review of scientific instruments 77, 103101 (2006).
+    """
+
+    def __name__(self):
+        return "ActiveCalibrationModel"
+
+    def __init__(
+        self,
+        driving_data,
+        force_voltage_data,
+        sample_rate,
+        bead_diameter,
+        driving_frequency_guess,
+        viscosity=1.002e-3,
+        temperature=20,
+        num_windows=5,
+    ):
+        """
+        Active Calibration Model.
+
+        This model fits data acquired in the presence of nanostage or mirror oscillation.
+
+        Parameters
+        ----------
+        driving_data : array_like
+            Array of driving data.
+        force_voltage_data : array_like
+            Uncalibrated force data in volts.
+        sample_rate : float
+            Sample rate at which the signals we acquired.
+        bead_diameter : float
+            Bead diameter [um].
+        driving_frequency_guess : float
+            Guess of the driving frequency.
+        viscosity : float, optional
+            Liquid viscosity [Pa*s].
+        temperature : float, optional
+            Liquid temperature [Celsius].
+        num_windows : int, optional
+            Number of windows to average for the uncalibrated force. Using a larger number of
+            windows potentially increases the bleed, but may be useful when the SNR is low.
+        """
```

Comment:
Hm, but then some of the input args wouldn't be documented anymore (since they don't end up as attributes). I'll add the attributes in the top docstring though.

Reply:
not a huge fan of all the duplicated documentation, but I do get your point. I think this is better since at least there's consistency between the two model types now